src/library/ResourceUtils/ResourceUtils.ts

```diff
@@ -84,7 +84,10 @@ export class ResourceUtils {
             const subPathValue = resourcePathValue[subPathIndex];
             if (this.isPrimitive(subPathValue)) {
```

Comment:
```suggestion
            if (this.isPrimitive(subPathValue) || pathSections.length === 2) {
```
this will perform the same way instead of another else if.